File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int Bprice = prices[0];
        int profit = 0;
        for (int i = 1;i < prices.size();i++){
            if (prices[i] < Bprice){
                Bprice = prices[i];
            }
            else{
                int currprofit = prices[i] - Bprice;
                profit = max(currprofit,profit);
            }
        }
        return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        Bprice = prices[0]
        profit = 0

        for i in range(len(prices)):
            if prices[i] <= Bprice:
                Bprice = prices[i]
            else:
                currprofit = prices[i] - Bprice
                profit = max(currprofit,profit)
        return profit 
